src/state/slices/opportunitiesSlice/resolvers/cosmosSdk/index.ts

```diff
@@ -0,0 +1,60 @@
+import { cosmosChainId, fromAccountId, osmosisChainId } from '@shapeshiftoss/caip'
+import type { Account, CosmosSdkBaseAdapter, CosmosSdkChainId } from '@shapeshiftoss/chain-adapters'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { logger } from 'lib/logger'
+import { isFulfilled, isRejected, isSome } from 'lib/utils'
+import type { ReduxState } from 'state/reducer'
+import { selectFeatureFlags, selectPortfolioAccounts } from 'state/slices/selectors'
+
+import type { GetOpportunityIdsOutput } from '../../types'
+import type { OpportunityIdsResolverInput } from '../types'
+import { makeUniqueValidatorAccountIds } from './utils'
+
+const moduleLogger = logger.child({ namespace: ['opportunities', 'resolvers', 'cosmosSdk'] })
+
+export const cosmosSdkOpportunityIdsResolver = async ({
+  reduxApi,
+}: OpportunityIdsResolverInput): Promise<{
+  data: GetOpportunityIdsOutput
+}> => {
+  const state = reduxApi.getState() as ReduxState
+
+  const { CosmosSdkOpportunitiesAbstraction } = selectFeatureFlags(state)
+  if (!CosmosSdkOpportunitiesAbstraction) {
+    return { data: [] }
+  }
+
+  const chainAdapters = getChainAdapterManager()
+  const portfolioAccountIds = Object.keys(selectPortfolioAccounts(state))
+  const cosmosSdkAccountIds = portfolioAccountIds.filter(accountId =>
+    [cosmosChainId || osmosisChainId].includes(fromAccountId(accountId).chainId),
```

Comment:
We can't just check the `chainNamespace`, since this includes Thorchain which doesn't use the regular Cosmos SDK staking module